src/api.v2/helpers/pipeline/handleQCResponse.js

```diff
@@ -0,0 +1,147 @@
+        await new Plot().upsert(
+          { id: plotUuid, experiment_id: experimentId },
+          { s3_data_key: s3DataKey },
+        )
+      ));
+
+    logger.log('Uploading plotData for', taskName, 'to DynamoDB');
+
+    // Promise.all stops if it encounters errors.
+    // This handles errors so that error in one upload does not stop others
+    await Promise.all(
+      plotConfigUploads.map((p) => p.catch((e) => logger.error(e))),
+    );
+  }
+};
+
+const updateProcessingConfig = async (taskName, experimentId, output, sampleUuid) => {
+  // TODO the processing config validation was not being enforced because the 'anyOf' requirement
+  // was not being correctly applied. This needs to be refactored together with the
+  // pipeline and ideally while unifying the qc & gem2s responses.
+  // await validateRequest(output, 'ProcessingConfigBodies.v1.yaml');
+
+  const experiment = new Experiment();
+
+  const {
+    processingConfig: previousConfig,
+  } = await experiment.findById(experimentId).first();
+
+  if (sampleUuid !== '') {
+    const { auto } = output.config;
+
+    // This is a temporary fix to save defaultFilterSettings calculated in the QC pipeline
+    // to patch for old experiments with hardcoded defaultFilterSettings.
+    // Remove this once we're done migrating to the new experiment schema with
+    // defaultFilterSettings
+    const sampleOutput = output;
+
+    if (auto) sampleOutput.config.defaultFilterSettings = output.config.filterSettings;
+
+    await experiment.updateProcessingConfig(experimentId, [
+      {
+        name: taskName,
+        body: {
+          ...previousConfig[taskName],
+          [sampleUuid]: { ...sampleOutput.config },
+        },
+      },
+    ]);
+  } else {
+    await experiment.updateProcessingConfig(experimentId, [
+      {
+        name: taskName,
+        body: output.config,
+      },
+    ]);
+  }
+};
+
+const sendUpdateToSubscribed = async (experimentId, message, output, error, io) => {
+  const statusRes = await getPipelineStatus(experimentId, constants.QC_PROCESS_NAME);
+  const statusResToSend = { pipeline: statusRes[constants.QC_PROCESS_NAME] };
+
+  // Concatenate into a proper response.
+  const response = {
+    ...message,
+    status: statusResToSend,
+    type: constants.QC_PROCESS_NAME,
+  };
+
+  if (output !== null) {
+    response.output = output;
+  }
+  if (error) {
+    logger.log(`Error in ${constants.QC_PROCESS_NAME} received`);
+    AWSXRay.getSegment().addError(error);
+  }
+  logger.log('Sending to all clients subscribed to experiment', experimentId);
+  io.sockets.emit(`ExperimentUpdates-${experimentId}`, response);
+};
+
+const handleQCResponse = async (io, message) => {
+  AWSXRay.getSegment().addMetadata('message', message);
+
+  await validateRequest(message, 'PipelineResponse.v1.yaml');
+
+  await hookRunner.run(message);
+
+  const { experimentId } = message;
+  const { error = false } = message.response || {};
+
+  let output = null;
```

Comment:
could you rename this `output` to `s3output` or `s3data`? Something less generic to indicate from its name immediately what it is about.

Reply:
did some renaming in the functions involved